### PR TITLE
Add sqlite lib

### PR DESCRIPTION
**Pull request by LionZXY**
#10

### CODE REVIEW

File: CMakeLists.txt [hunk 1]
```diff
@@ -18,3 +21,4 @@ add_executable(Recast-server ${SOURCE_FILES})
 
 target_link_libraries(Recast-server ${Boost_LIBRARIES})
 target_link_libraries(Recast-server ${CMAKE_THREAD_LIBS_INIT})
+target_link_libraries(Recast-server sqlite3)
```
Comment (StealthTech) [inline CMakeLists.txt:25]: There's no newline at eof

File: CMakeLists.txt [hunk 2]
```diff
@@ -3,6 +3,9 @@ project(Recast-server)
 
 set(CMAKE_CXX_STANDARD 14)
 
+add_subdirectory(libs)
+include_directories(libs)
+
 file(GLOB_RECURSE SOURCE_FILES
         "src/*.h"
```
Comment (StealthTech) [inline CMakeLists.txt:10]: Don't we use hpp headers?

Comment (LionZXY) [reply CMakeLists.txt:10]: This is mistake

File: libs/sqlite/CMakeLists.txt [hunk 3]
```diff
@@ -0,0 +1,19 @@
...
+set_target_properties(sqlite PROPERTIES OUTPUT_NAME sqlite3)
+
+add_definitions(-DSQLITE_ENABLE_RTREE)
+add_definitions(-DSQLITE_ENABLE_FTS4)
+add_definitions(-DSQLITE_ENABLE_JSON1)
+
+target_link_libraries(sqlite3 ${CMAKE_DL_LIBS})
+install(FILES src/sqlite3.h src/sqlite3ext.h DESTINATION include)
+install(TARGETS sqlite3 LIBRARY DESTINATION lib ARCHIVE DESTINATION lib)
+install(TARGETS sqlite RUNTIME DESTINATION bin)
```
Comment (StealthTech) [inline libs/sqlite/CMakeLists.txt]: No newline at eof again

File: libs/sqlite/README.md [hunk 4]
```diff
@@ -0,0 +1,24 @@
...
+```
+  cmake -G "Visual Studio 12 Win64" -DCMAKE_INSTALL_PREFIX=d:/libs
+  < open and build sqlite3.sln >
+```
+
+On Linux/FreeBSD without root access
+```
+  cmake -DCMAKE_BUILD_TYPE=Release -DCMAKE_INSTALL_PREFIX=/home/student/libs
+  make install
+```
```
Comment (StealthTech) [inline libs/sqlite/README.md]: Do we need this readme in our repository?

File: libs/sqlite/CMakeLists.txt (hunk 3 above)
Comment (LionZXY) [reply libs/sqlite/CMakeLists.txt]: What?

File: libs/sqlite/appveyor.yml [hunk 5]
```diff
@@ -0,0 +1,50 @@
...
+  - 7z a c:\projects\sqlite\sqlite.zip * -tzip
+  - cd c:\projects\sqlite
+
+artifacts:
+  - path: sqlite.zip
+    name: sqlite.zip
+
+deploy_script:
+  - cd c:\projects\sqlite
+  - curl -T sqlite.zip --user %ACCOUNT% https://webdav.yandex.ru/libs/sqlite.zip
```
Comment (StealthTech) [inline libs/sqlite/appveyor.yml]: I think we don't need this

File: src/Main.cpp [hunk 6]
```diff
@@ -7,13 +7,15 @@
...
  **/
+#include <sqlite/src/sqlite3.h>
 #include "Server.h"
 
 /**
  * @brief Main method :)
  **/
 int main() {
     Server server;
+    sqlite3 *sq;
```
Comment (StealthTech) [inline src/Main.cpp]: Is this for future?

**Review by StealthTech — APPROVED** [review]

File: src/Main.cpp (hunk 6 above)
Comment (LionZXY) [reply src/Main.cpp]: Ooops...

**Pull request merged**